src/contracts/facilitators/gsm/GhoReserve.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IERC20} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
+import {VersionedInitializable} from '@aave/core-v3/contracts/protocol/libraries/aave-upgradeability/VersionedInitializable.sol';
+import {IGhoReserve} from './interfaces/IGhoReserve.sol';
+
+/**
+ * @title GhoReserve
+ * @author Aave/TokenLogic
+ * @notice GHO Remote Reserve. It provides withdraw/repay facilities to a GHO Stability Module in order to provide GHO liquidity on a remote chain.
+ * @dev To be covered by a proxy contract.
+ */
+contract GhoReserve is IGhoReserve, Ownable, VersionedInitializable {
+  /// @inheritdoc IGhoReserve
+  address public immutable GHO_TOKEN;
+
+  /// @dev Mapping to keep track of GHO withdrawn by an address and capacity
+  mapping(address => GhoCapacity) private _ghoCapacity;
+
+  /**
+   * @dev Constructor
+   * @param ghoAddress Address of GHO token on the remote chain
+   */
+  constructor(address ghoAddress) Ownable() {
+    require(ghoAddress != address(0), 'INVALID_ZERO_ADDRESS');
+
+    GHO_TOKEN = ghoAddress;
+  }
+
+  /**
+   * @notice GhoReserve initializer
+   * @param admin The address of the default admin role
+   */
+  function initialize(address admin) external initializer {
```

Comment:
```suggestion
  /**
   * @dev Initializer
   * @param owner The address of the owner
   */
  function initialize(address owner) external initializer {
```
or an alternative like `newOwner` in case of shadowing